weave/integrations/verifiers/verifiers.py

```diff
@@ -25,6 +32,84 @@ def _verifiers_postprocess_inputs(inputs: dict[str, Any]) -> dict[str, Any]:
     return inputs
 
 
+def _remove_logprobs_from_responses(responses: list[Any]) -> list[Any]:
+    """Return a copy of responses with logprobs removed, without mutating originals."""
+    sanitized: list[Any] = []
+    for response in responses:
+        if isinstance(response, ModelResponse):
+            # Prefer pydantic v2 deep copy if available; fall back to deepcopy
+            try:
+                response_copy = response.model_copy(deep=True)  # type: ignore[attr-defined]
+            except Exception:
+                response_copy = copy.deepcopy(response)
+            # Strip logprobs from the copied response
+            for choice in getattr(response_copy, "choices", []) or []:
+                if hasattr(choice, "logprobs"):
+                    choice.logprobs = None
+            sanitized.append(response_copy)
+        else:
+            sanitized.append(response)
+    return sanitized
+
+
+def _verifiers_postprocess_outputs_no_logprobs(outputs: Any) -> Any:
+    """Return a logging-safe copy of outputs with logprobs removed.
+
+    IMPORTANT: Do not mutate the original outputs, as they are consumed by the
+    training pipeline which expects `logprobs` to be present.
+    """
+    if outputs is None:
+        return outputs
+
+    # Deep copy the entire outputs structure for logging only
+    try:
+        outputs_copy = (
+            outputs.model_copy(deep=True)  # type: ignore[attr-defined]
+            if isinstance(outputs, BaseModel)
+            else copy.deepcopy(outputs)
+        )
+    except Exception:
+        outputs_copy = copy.deepcopy(outputs)
```

Comment:
why not just deepcopy?  What is the point of this?

Reply:
Ayush mentioned `model_copy` might be more performant? is that true